### PR TITLE
DP-663: Reduce cyclomatic complexity `validateSchema`

### PR DESCRIPTION
**Pull request by cnemarich**
### Summary of Proposed Changes
Reduces the cyclomatic complexity of the function `validateSchema` to lower than 15 to stop tripping linter.
* Combines all OPU tests under one if-then statement
* Reorders checks for `schema` and `template_path` to minimize `if-then` branching
* Removes `if-then` checks for failures on each test in favor of checking all tests at the end and removing any with zero rows
* Reduces overall cyclomatic complexity from 22 to 13

### Related Issues
- DP-663

## Reviewer:
- [x] Tests added/updated & passing.
- [x] Clean linting.
- [x] Related issue ticket in Jira/GitHub.
- [x] Documentation added/updated.
- [x] Code conforms to style guidelines.
- [x] Well commented.
- [ ] Updates reflected in NEWS.md.
- [x] Build check passes.

### CODE REVIEW

File: R/unPackSchema.R [hunk 1]
```diff
@@ -40,36 +40,52 @@ validateSchema <- function(schema,
     assign(p, purrr::pluck(params, p))
   }
 
-  # If template_path provided, check it and unpack it to create comparison schema.
+  # Checks to perform if schema_provided is TRUE ####
+  if (schema_provided) {
```
Comment (jason-p-pickering) [inline R/unPackSchema.R:44]: My preference would be to move this to a small unit tested function instead. Can you move it out of the main function?

File: R/unPackSchema.R [hunk 2]
```diff
@@ -40,36 +40,52 @@ validateSchema <- function(schema,
...
+    schema %<>% check_schema(schema = .,
+                             cop_year = cop_year,
+                             tool = tool,
+                             season = season)
+  }
+
+
+  # Checks to perform if filepath_provided is TRUE ####
   if (filepath_provided) {
+    ## If template_path provided, check it and unpack it to create comparison schema.####
```
Comment (jason-p-pickering) [inline R/unPackSchema.R:55]: Same here. I think this can be moved to a small function instead.

File: R/unPackSchema.R [hunk 3]
```diff
@@ -40,36 +40,52 @@ validateSchema <- function(schema,
...
 
     filepath_schema <-
       unPackSchema_datapack(
         template_path = template_path,
         skip = skip_tabs(tool = tool, cop_year = cop_year),
         cop_year = cop_year)
 
-  # If schema_object not provided, use filepath_schema
+    ## If schema_object is provided, check schema against filepath_schema ####
+    if (schema_provided && !identical(schema, filepath_schema)) {
```
Comment (jason-p-pickering) [inline R/unPackSchema.R:68]: Same here. Another small function.

**Review by jason-p-pickering — CHANGES_REQUESTED** [review]
I think this is a good start, but I think we can do more to pull out some of the conditional code into separate small units which we can test.

**Comment by cnemarich** [comment]
Function names are too long and need to be shortened in a number of cases

**Review by jason-p-pickering — APPROVED** [review]

**Comment by jason-p-pickering** [comment]
@JordanBalesBAO can you have a look at this PR? Both @cnemarich and I have worked on it, so better that we get someone else to review it.

**Comment by JordanBalesBAO** [comment]
@jason-p-pickering Sure!

**Review by JordanBalesBAO — APPROVED** [review]
Looks good!

**Pull request merged**